### DISCUSSION TITLE
Code coverage on getParam

### DISCUSSION THREAD

**Discussion by WilkAndy** [idaholab/moose#21795] [Q&A Tools] [answered]
Hi eveyrone,

In https://mooseframework.inl.gov/docs/PRs/21592/coverage/geochemistry/src/userobjects/GeochemistryKineticRate.C.gcov.html you can see that the final line in

```
     133         121 :     _monod_ind(isParamValid("promoting_monod_indices")
     134             :                    ? getParam<std::vector<Real>>("promoting_monod_indices")
     135           0 :                    : std::vector<Real>(_promoting_names.size(), 0.0)),
```

is not covered by any test.  However, i have tests that supply `promoting_monod_indices` and tests that do not supply that parameter.  Why is the final line not covered by the tests?

a

**Comment by loganharbour** [comment, marked as answer]
In order for us to afford running coverage with so many different configurations and tests, all of the module coverage is produced in `opt` mode (with `-O2`, lots of optimization). This optimization... can unfortunately trick `gcov`. This is likely one of those cases. 

You're probably not going to be able to hit 100% with this. Rest assured, you know that it's tested 😆

**Reply by WilkAndy** [reply]
Thanks you @loganharbour
